darts/models/forecasting/moving_average.py

```diff
@@ -0,0 +1,35 @@
+"""
+Moving Average
+-------------------------------
+"""
+import numpy as np
+
+from darts.models.forecasting.forecasting_model import ForecastingModel
+from darts.timeseries import TimeSeries
+
+
+class MovingAverageForecaster(ForecastingModel):
+    def __init__(self, window: int):
+        """Moving Average Model
+
+        Parameters
+        ----------
+        window
+            The length of the window over which to average values.
+        """
+        super().__init__()
+        self.window = window
+        self.mean_val = None
+
+    def __str__(self):
+        return "Moving average predictor model"
+
+    def fit(self, series: TimeSeries):
+        super().fit(series)
+        self.mean_val = series[-self.window :].univariate_values().mean()
+        return self
+
+    def predict(self, n: int, num_samples: int = 1):
+        super().predict(n, num_samples)
+        forecast = np.array([self.mean_val for _ in range(n)])
```

Comment:
I think we should rather do something where we constantly take the mean of the last `window` values. I.e., for `i = 0, ..., n-1` we would have something like
* `val_0 = mean(series[-window:])`
* `val_1 = mean(series[-window+1:] + [val_0])`
etc...

So we wouldn't return a constant value, which I think would be a more natural interpretation of what a moving average is. Unless I'm maybe missing something.
Also it'd be better to Numpy-vectorize all this code as much as we can :)

Reply:
This would be a nicer way to avoid returning a constant value. But I'm not entirely sure if it can be vectorized since it's a recurrent computation. I will try take a look at it.

Reply:
Note that your current computation is also not vectorized, as it relies on a Python list comprehension. It's OK to have a loop over `n` if there's no clean way to vectorize.